### PR TITLE
Feat / accessibility improvements and quick wins

### PR DESCRIPTION
**Pull request by MelissaDTH**
## This PR functions as a feature branch

Any quick wins that were found during assessment of the WCAG 2.1 guidelines, are documented and can be found: [here](https://docs.google.com/document/d/1UgX2ffXBXcIujBwW1DxLOkHDwGB_lOeh8dcH8iDmhwg/edit#heading=h.ua3zqyh0w12m). The following of these have been implemented:

- Corrects the heading structure throughout the app
- Skiplink in header

The following features will be added with PR #6 
- Adds an underline to links at all times
- Adjust the Page title of the Home page
- Prepares the Modal and Dialog components for the role attribute
    - correct roles will be added in the next commit

### CODE REVIEW

File: src/components/Account/Account.tsx [hunk 1]
```diff
@@ -168,6 +168,7 @@ const Account = ({ panelClassName, panelHeaderClassName, canUpdateEmail = true }
             onSubmit: (values) => updateUser({ firstName: values.firstName || '', lastName: values.lastName || '' }),
             content: (section) => (
               <>
+                <h1 className={styles.hideUntilFocus}>{t('nav.account')}</h1>
```
Comment (royschut) [inline src/components/Account/Account.tsx]: Are we sure about this h1? Although 'my account' is a different route, on the screen it is actually a segment of the User screen, with a user menu as well. Should this be an h2 and should we have a User h1?

File: src/components/RegistrationForm/RegistrationForm.tsx [hunk 2]
```diff
@@ -117,7 +117,7 @@ const RegistrationForm: React.FC<Props> = ({
         <Checkbox
           key={index}
           name={consent.name}
-          value={consent.value || ''}
+          value={''}
```
Comment (royschut) [inline src/components/RegistrationForm/RegistrationForm.tsx]: Mistake I think?

File: src/components/ResetPasswordForm/ResetPasswordForm.tsx [hunk 3]
```diff
@@ -14,8 +14,10 @@ type Props = {
 const ResetPasswordForm: React.FC<Props> = ({ onCancel, onReset, submitting }: Props) => {
   const { t } = useTranslation('account');
   return (
-    <div className={styles.resetPassword}>
-      <h5 className={styles.title}>{t('reset.reset_password')}</h5>
+    <div className={styles.resetPassword} role="dialog" aria-labelledby="resetPassword">
+      <h1 id="resetPassword" className={styles.title}>
+        {t('reset.reset_password')}{' '}
```
Comment (royschut) [inline src/components/ResetPasswordForm/ResetPasswordForm.tsx]:
```suggestion
        {t('reset.reset_password')}
```

File: src/components/ResetPasswordForm/ResetPasswordForm.tsx [hunk 4]
```diff
@@ -14,8 +14,10 @@ type Props = {
 const ResetPasswordForm: React.FC<Props> = ({ onCancel, onReset, submitting }: Props) => {
   const { t } = useTranslation('account');
   return (
-    <div className={styles.resetPassword}>
-      <h5 className={styles.title}>{t('reset.reset_password')}</h5>
+    <div className={styles.resetPassword} role="dialog" aria-labelledby="resetPassword">
+      <h1 id="resetPassword" className={styles.title}>
```
Comment (royschut) [inline src/components/ResetPasswordForm/ResetPasswordForm.tsx]: Minor, but snake_case is most preferable for id's: `reset_password`

File: src/components/VideoDetailsInline/VideoDetailsInline.tsx [hunk 5]
```diff
@@ -20,7 +20,7 @@ const VideoDetailsInline: React.FC<Props> = ({ title, description, primaryMetada
   const breakpoint: Breakpoint = useBreakpoint();
   const isMobile = breakpoint === Breakpoint.xs;
 
-  const TitleComponent = typeof title === 'string' ? 'h2' : 'div';
+  const TitleComponent = typeof title === 'string' ? 'h1' : 'div';
```
Comment (royschut) [inline src/components/VideoDetailsInline/VideoDetailsInline.tsx]: Is this indeed the title of the whole page?

File: src/styles/accessibility.scss [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+// Instead, we apply styles to the focus state first, which solves the purpose, but unwantedly also for the active state
+// We fix this for non-Safari browsers afterwards by resetting to default and then re-applying using focus-visible
+@mixin focus-visible($value: $focus-box-shadow, $default-value: none) {
+  @media (hover: hover) and (pointer: fine) {
+    &:not(:disabled) {
+      &:focus {
+        @content ($value);
+      }
+      &:focus:not(:focus-visible) {
+        @content ($default_value);
```
Comment (royschut) [inline src/styles/accessibility.scss]:
```suggestion
        @content ($default-value);
```

**Review by royschut — COMMENTED** [review]
Nice Melissa!
I have a few considerations:

File: src/components/ChooseOfferForm/ChooseOfferForm.tsx [hunk 7]
```diff
@@ -81,7 +81,7 @@ const ChooseOfferForm: React.FC<Props> = ({
         aria-label={ariaLabel}
       />
       <label className={styles.label} htmlFor={offer.offerId}>
-        <h4 className={styles.offerTitle}>{title}</h4>
+        <h2 className={styles.offerTitle}>{title}</h2>
```
Comment (langemike) [inline src/components/ChooseOfferForm/ChooseOfferForm.tsx]: I agree with this h2 change, but I noticed a parent <label> containing a whole lot of HTML. I think the `label` should be `role="option"` or something similar. Not a quickwin, but make sure you have wrote it down in your list because this should use some optimisation ;)

File: src/components/RegistrationForm/RegistrationForm.tsx (hunk 2 above)
Comment (langemike) [inline src/components/RegistrationForm/RegistrationForm.tsx:120]: Why is this change needed?

File: src/components/ResetPasswordForm/ResetPasswordForm.tsx [hunk 8]
```diff
@@ -14,8 +14,10 @@ type Props = {
 const ResetPasswordForm: React.FC<Props> = ({ onCancel, onReset, submitting }: Props) => {
   const { t } = useTranslation('account');
   return (
-    <div className={styles.resetPassword}>
-      <h5 className={styles.title}>{t('reset.reset_password')}</h5>
+    <div className={styles.resetPassword} role="dialog" aria-labelledby="resetPassword">
```
Comment (langemike) [inline src/components/ResetPasswordForm/ResetPasswordForm.tsx]: I would have chosen an id of `modalTitle` or something similar. But if multiple occurrences of this (modal) HTML can exist in the DOM, than I would definitely make it unique. No need for a change, just want to make you aware of this.

File: src/components/RegistrationForm/RegistrationForm.tsx (hunk 2 above)
Comment (MelissaDTH) [reply src/components/RegistrationForm/RegistrationForm.tsx]: I cannot explain how this slipped through, is indeed a mistake and was solely for testing purposes! Thanks 😄

**Review by langemike — APPROVED** [review]
The solutions you chosen seem logical to me! Of course we should understand, that some heading level changes requires context to determine the accurate level. 

This is something Roy also noted about 'my account'.

I noticed a `<label>` - an offer choice - which contains a lot of HTML. Which I think is not good from an accessibility-standpoint, but that's not something for now to fix ;)

I also believe @royschut placed his comments using the Review-mechanism. Therefore there were some double comments. I deleted one of them. ;-)

File: src/components/ChooseOfferForm/ChooseOfferForm.tsx (hunk 7 above)
Comment (MelissaDTH) [reply src/components/ChooseOfferForm/ChooseOfferForm.tsx]: Thanks! Added it to the checklist

File: src/components/VideoDetailsInline/VideoDetailsInline.tsx (hunk 5 above)
Comment (MelissaDTH) [reply src/components/VideoDetailsInline/VideoDetailsInline.tsx]: Yes, this is the title that's displayed underneath the Player:
Screenshot 2023-10-19 at 15 44 59

File: src/components/Account/Account.tsx (hunk 1 above)
Comment (MelissaDTH) [reply src/components/Account/Account.tsx]: Hmm, but in the user menu this is also called 'Account', so for the user I would say this is more clear. Otherwise we will have a 'User' heading, what information does this convey? I'm not convinced yet that it's clear enough... 🤔

File: src/components/VideoDetailsInline/VideoDetailsInline.tsx (hunk 5 above)
Comment (royschut) [reply src/components/VideoDetailsInline/VideoDetailsInline.tsx]: I see, makes sense!

File: src/components/Account/Account.tsx (hunk 1 above)
Comment (royschut) [reply src/components/Account/Account.tsx]: Hm, I understand your doubt. Let's go for your approach for now, we could always reconsider!

**Review by royschut — APPROVED** [review]
:accessibility: 💪

File: src/components/ResetPasswordForm/ResetPasswordForm.tsx (hunk 8 above)
Comment (MelissaDTH) [reply src/components/ResetPasswordForm/ResetPasswordForm.tsx]: Thanks, that is good feedback! On this page, in some configs, multiple dialogs can occur in the DOM. So on this page it's good practice

File: src/pages/Home/Home.tsx [hunk 9]
```diff
@@ -1,5 +1,8 @@
 import React from 'react';
 import shallow from 'zustand/shallow';
+import { useTranslation } from 'react-i18next';
+
+import '../../styles/accessibility.scss';
```
Comment (ChristiaanScheermeijer) [inline src/pages/Home/Home.tsx]: I think this import can be moved to be in the main SCSS file. It currently creates a global rule that we only should define once. We probably want to split the global selector from the module selector:

```tsx
import styles from './Home.module.scss';

return <div className={styles.hideUntilFocus} />
```

VS

```tsx
import '../../styles/accessibility.scss';

return <div className="hideUntilFocus" />
```

**Review by royschut — CHANGES_REQUESTED** [review]
I think your last update breaks some hidden h1 tags? You've moved the `hideUntilFocus` class to a page specific css file, but it should be available everywhere. I think what Christiaan means is that we don't import the `accessiblity.scss` file from the Home screen, but from the main scss file and then use it in the Home styling.

**Comment by MelissaDTH** [comment]
> I think your last update breaks some hidden h1 tags? You've moved the `hideUntilFocus` class to a page specific css file, but it should be available everywhere. I think what Christiaan means is that we don't import the `accessiblity.scss` file from the Home screen, but from the main scss file and then use it in the Home styling.

Thanks, I misunderstood. I've pushed a new commit regarding your feedback but still have some failing tests because I did change the hierarchy of the header elements in this PR, I will fix that today!

File: src/components/Account/__snapshots__/Account.test.tsx.snap [hunk 10]
```diff
@@ -8,14 +8,19 @@ exports[`<Account> > renders and matches snapshot 1`] = `
...
     </div>
     <form
       class="_flexBox_1c1c63"
       novalidate=""
     >
+      <h1
+        class="hideUntilFocus"
+      >
+        nav.account
+      </h1>
```
Comment (royschut) [inline src/components/Account/__snapshots__/Account.test.tsx.snap]: This is kind of a strange snapshot. Can it happen that an h2 appears first, before the h1? Is the h1 perhaps placed incorrectly?

File: src/components/Account/Account.tsx [hunk 11]
```diff
@@ -6,6 +6,7 @@ import DOMPurify from 'dompurify';
 import { useMutation } from 'react-query';
 
 import styles from './Account.module.scss';
+import '../../styles/accessibility.scss';
```
Comment (royschut) [inline src/components/Account/Account.tsx]: This import isn't needed, since you're importing it from the scss file as well.

File: src/components/ChooseOfferForm/__snapshots__/ChooseOfferForm.test.tsx.snap [hunk 12]
```diff
@@ -257,16 +257,16 @@ exports[`<OffersForm> > renders and matches snapshot 2`] = `
...
     novalidate=""
   >
-    <h2
+    <h1
       class="_title_f7961f"
     >
       choose_offer.title
-    </h2>
-    <h3
+    </h1>
```
Comment (royschut) [inline src/components/ChooseOfferForm/__snapshots__/ChooseOfferForm.test.tsx.snap]: This is the second `h1` in this snapshot. How can that be?

File: src/containers/Layout/Layout.tsx [hunk 13]
```diff
@@ -188,7 +188,9 @@ const Layout = () => {
           <hr className={styles.divider} />
           {renderUserActions()}
         </Sidebar>
-        <Outlet />
+        <div id="content" className={styles.content} tabIndex={-1}>
+          <Outlet />
+        </div>
```
Comment (royschut) [inline src/containers/Layout/Layout.tsx]: Maybe for later, but perhaps this should be a `<main />` landmark. @langemike?

File: src/pages/User/__snapshots__/User.test.tsx.snap [hunk 14]
```diff
@@ -107,14 +107,19 @@ exports[`User Component tests > Account Page 1`] = `
...
         </div>
         <form
           class="_flexBox_1c1c63"
           novalidate=""
         >
+          <h1
+            class="hideUntilFocus"
+          >
+            nav.account
+          </h1>
```
Comment (royschut) [inline src/pages/User/__snapshots__/User.test.tsx.snap]: Also here, an `h1` after an `h2`. Is this correct?

File: src/pages/User/__snapshots__/User.test.tsx.snap [hunk 15]
```diff
@@ -369,9 +374,9 @@ exports[`User Component tests > Favorites Page 1`] = `
         <div
           class="_header_546956"
         >
-          <h3>
+          <h1>
             favorites.title
-          </h3>
+          </h1>
```
Comment (royschut) [inline src/pages/User/__snapshots__/User.test.tsx.snap]: And a double one.

File: src/components/Payment/Payment.tsx [hunk 16]
```diff
@@ -7,6 +7,7 @@ import ExternalLink from '../../icons/ExternalLink';
 import IconButton from '../IconButton/IconButton';
 
 import styles from './Payment.module.scss';
+import '../../styles/accessibility.scss';
```
Comment (royschut) [inline src/components/Payment/Payment.tsx]: Maybe move it to the scss file, and use `src/` import?

**Review by royschut — COMMENTED** [review]
Sorry, I have a few more considerations. I hope they're small, but let's Slack in case you want to move on!

File: src/components/Account/Account.tsx (hunk 11 above)
Comment (MelissaDTH) [reply src/components/Account/Account.tsx]: Thanks!

File: src/components/Account/__snapshots__/Account.test.tsx.snap (hunk 10 above)
Comment (MelissaDTH) [reply src/components/Account/__snapshots__/Account.test.tsx.snap]: Sharp! In some occasions, depending if you're on a SVOD or AUTHVOD config and depending of the value of the variable `isChangingOffer`, one of the Payment element is shown or not. I've used that element to determine where the overarching `<h1>` element should be. Thanks!

I'm not entirely sure how it can be that the `h2` element is positioned after the `h1` element in this snapshot, but I've checked that case with the `h2` "account.about_you" element and the hidden h1` `element in the app, and they are positioned correctly. 

I also wanna add that this is the snapshot file for the whole User page, which includes the Account subpage, Payment subpage and Favorite subpage. I think this explains why the hierarchy is not completely correct.

File: src/pages/User/__snapshots__/User.test.tsx.snap (hunk 14 above)
Comment (MelissaDTH) [reply src/pages/User/__snapshots__/User.test.tsx.snap]: See my comment here: https://github.com/Videodock/ott-web-app/pull/3#discussion_r1430288079 😃

File: src/pages/User/__snapshots__/User.test.tsx.snap (hunk 15 above)
Comment (MelissaDTH) [reply src/pages/User/__snapshots__/User.test.tsx.snap]: This is correct, on the Favorites page we do want this element to become the `h1` element, and since this snapshot file includes everything on the User page (like I said in https://github.com/Videodock/ott-web-app/pull/3#discussion_r1430288079, I think that is why you see the double `h1` element

File: src/components/Account/__snapshots__/Account.test.tsx.snap (hunk 10 above)
Comment (royschut) [reply src/components/Account/__snapshots__/Account.test.tsx.snap]: Ah yes. So then it is correct? It is not rendered as such in the DOM? (An Axetool check would notice it)

File: src/pages/User/__snapshots__/User.test.tsx.snap (hunk 15 above)
Comment (royschut) [reply src/pages/User/__snapshots__/User.test.tsx.snap]: I see, in practise they are never rendered at the same time, all good! :-)

File: src/pages/User/__snapshots__/User.test.tsx.snap (hunk 14 above)
Comment (royschut) [reply src/pages/User/__snapshots__/User.test.tsx.snap]: Yes :)

File: src/components/Account/__snapshots__/Account.test.tsx.snap (hunk 10 above)
Comment (royschut) [reply src/components/Account/__snapshots__/Account.test.tsx.snap]: Yes it is :)

File: src/components/Payment/Payment.tsx [hunk 17]
```diff
@@ -170,7 +170,8 @@ const Payment = ({
       {accessModel === ACCESS_MODEL.SVOD && (
         <div className={panelClassName}>
           <div className={panelHeaderClassName}>
-            <h3>{isChangingOffer ? t('user:payment.change_plan') : t('user:payment.subscription_details')}</h3>
+            {!isChangingOffer && <h1 className="hideUntilFocus">{t('nav.payments')}</h1>}
```
Comment (royschut) [inline src/components/Payment/Payment.tsx]: Why is this conditional? We always need the hidden h1 right?

File: src/components/Payment/Payment.tsx [hunk 18]
```diff
@@ -272,7 +273,8 @@ const Payment = ({
       )}
       <div className={panelClassName}>
         <div className={panelHeaderClassName}>
-          <h3>{t('user:payment.payment_method')}</h3>
+          {!isChangingOffer && <h1 className="hideUntilFocus">{t('nav.payments')}</h1>}
```
Comment (royschut) [inline src/components/Payment/Payment.tsx]: Isn't this the second h1?

**Review by royschut — COMMENTED** [review]
I apologise for not seeing them earlier, but I have more questions/comments:

File: src/containers/Layout/Layout.tsx (hunk 13 above)
Comment (langemike) [reply src/containers/Layout/Layout.tsx]: I noticed there are already more occurrences of `<main>` within the app.

I think it's better to apply `<main>` landmark per every page so we have the most flexibility. Eg. search page with filters on the left side, a big (jumbotron) header with a card grid below it etc, user setting page with a panel with links on the side etc.

Yes that means a little bit repetition, but that is the way it is.

We need to create a ticket for this, because not every page currently has a `<main>` landmark.

File: src/components/Payment/Payment.tsx (hunk 17 above)
Comment (langemike) [reply src/components/Payment/Payment.tsx]: Your feedback seems logical Roy, but I have to see the whole context before i can make that assumption. Good call, though.

Comment (MelissaDTH) [reply src/components/Payment/Payment.tsx]: @royschut Yes we do! Just like I explained in a previous comment:
In some occasions, depending if you're on a SVOD or AUTHVOD config and depending of the value of the variable `isChangingOffer`, one of the Payment element is shown or not. I've used the `isChangingOffer` variable to determine where the hidden `<h1>` element should be.

File: src/components/Payment/Payment.tsx (hunk 18 above)
Comment (MelissaDTH) [reply src/components/Payment/Payment.tsx]: See https://github.com/Videodock/ott-web-app/pull/3#discussion_r1434917952

File: src/containers/Layout/Layout.tsx (hunk 13 above)
Comment (MelissaDTH) [reply src/containers/Layout/Layout.tsx]: Discussed this with Mike: This can stay a div, but every page does need a `<main>` landmark, we will create a ticket to work on this the upcoming sprint!

**Review by royschut — APPROVED** [review]
